### PR TITLE
MULTIARCH-4315: Implements workloads partitioning annotations

### PR DESCRIPTION
**Pull request by aleskandro**
See https://github.com/openshift/enhancements/blob/c5b9aea25e2a7afd5f460c6a0eca7584b1685ce3/enhancements/workload-partitioning/management-workload-partitioning.md

Related to https://github.com/openshift/release/pull/51275

### CODE REVIEW

**Comment by Prashanth684** [comment]
Do we need e2e for these ? i would like to see some performance related testing on SNO and the like to really qualify this use case.

**Comment by aleskandro** [comment]
> Do we need e2e for these ? i would like to see some performance related testing on SNO and the like to really qualify this use case.

I think the operator still qualifies for the use case of sno + multiple workers in the edge scenarios in particular. I also want to read some data asap to better tune some choices we've made at first.

**Comment by aleskandro** [comment]
/override "Red Hat Konflux / multiarch-tuning-operator-enterprise-contract / multiarch-tuning-operator" "Red Hat Konflux / multiarch-tuning-operator-enterprise-contract / multiarch-tuning-operator-bundle" 

(we just need new snapshots)

File: bundle/manifests/multiarch-tuning-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -270,9 +272,6 @@ spec:
                   initialDelaySeconds: 5
                   periodSeconds: 10
                 resources:
-                  limits:
```
Comment (Prashanth684) [inline bundle/manifests/multiarch-tuning-operator.clusterserviceversion.yaml:273]: why remove the limits?

Comment (aleskandro) [reply bundle/manifests/multiarch-tuning-operator.clusterserviceversion.yaml:273]: it's a requirement from origin to have all the pods in openshift-* to be in the `burstable` QoS class

Comment (Prashanth684) [reply bundle/manifests/multiarch-tuning-operator.clusterserviceversion.yaml:273]: right right..or is it besteffort? but yes, got it

**Pull request merged**